src/ImageSharp/Processing/Processors/Binarization/BinaryThresholdProcessor.cs

```diff
@@ -19,6 +19,12 @@ public enum BinaryThresholdColorComponent : int
         /// Saturation color component as value to be compared to threshold.
         /// </summary>
         Saturation = 1,
+
+        /// <summary>
+        /// Colorfulness (saturated with min luminance 10%) color component as value to be
+        /// compared to threshold.
+        /// </summary>
+        Colorfulness_L10 = 2,
```

Comment:
This isn't part of the original specification.

Reply:
Ok, I see your point. I would see a certain degree of general purpose because saturation is really a problem at the dark end of the color space. At the light end saturation is always low, but rgb(0,1,0) has high saturation though it's practically black. The proposed Colorfulness_L10 works well in our toad application, and different L thresholds might be realised. A generalization with L threshold as argument would be possible but perhaps less convenient for users.

I fully understand if you don't want to have this in your lib, but we do need it. Would you recommend a fork? or is there a way to make extensions like that without integrating it?

Thanks and cheers,
Franz

Reply:
I don't like the idea of introducing enum-driven arbitrary strategies (they may be useless for most users), but we can generalize the feature by passing something like a `Func<Rgba32,byte,bool>` or `Func<Vector4,float,bool>` or equivalent to the processor. It's not gonna be trivial work, but I'm happy to guide you with such a contribution if you're interested.

Alternatively, you can just copy the code of `BinaryThresholdProcessor` and `BinaryThresholdProcessor<T>` into your application, and apply the changes there. Forking the entire project is unnecessary.

Reply:
Apologies for the silence. Day job is eating up all my waking hours. 

I think even a func overload would have to be very carefully designed so it wasn’t a performance killer and feels very niche. I’m happy with the general enum but specific cases can be delivered like Anton said. The library is designed to be extensible.

Reply:
First I want to thank you for your time spent on this PR, both of you.

My arbitrarily defining a 'Colorfulness_L10' color component was too pragmatic. I will have a closer look on color standards like BT.709 or JPEG YCbCr. We need a scalar value to compare to a threshold. This value has to represent colorfulness or chromaticity to the human eye for my purposes. The HSL saturation value comes near but fails for dark colors. The distance of Cb or Cr to the achromatic level AL would be a candidate.

    Chroma = Max(Abs(Cb - AL), Abs(Cr - AL))

Would that be general purpose in your eyes?

Reply:
I like the approach but my understanding of colorspaces is not good enough not to evaluate the proposal.

Quick sanity test: How would you name this "mode"? How would you explain it in docs? Have you tested it already?

Reply:
I think I understand the software requirements better now and will investigate the issue more thoroughly first. When I have a suitable solution from my point of view, I will make a corresponding proposal. Until then I will work on a branch without PRs. Maybe I will do a merge to saturation_threshold from time to time to see the results of the tests. Please just ignore and let the PR stay open! I will get back to you when I have a state that is ok from my point of view.